### PR TITLE
Update package:test dependencies to `0.12.18+2`

### PR DESCRIPTION
**Pull request by tvolkert**

### CODE REVIEW

**Comment by tvolkert** [comment]
@pq

**Comment by pq** [comment]
Assuming Travis is happy, LGTM!

**Comment by pq** [comment]
Sounds like the resolution is to wait for `0.12.19+1` ?

**Comment by tvolkert** [comment]
> Sounds like the resolution is to wait for `0.12.19+1`?

Yep, there were analyzer errors in `0.12.18+1` (fixed in https://github.com/dart-lang/test/pull/518 but not published in `0.12.18`), so I'm moving straight to `0.12.19+1`

File: packages/flutter/test/painting/text_style_test.dart [hunk 1]
```diff
@@ -14,7 +14,8 @@ void main() {
       fontWeight: FontWeight.w800,
       height: 123.0,
     );
-    expect(() { s1.fontFamily = 'test'; }, throws); // ignore: ASSIGNMENT_TO_FINAL
+    // ignore: ASSIGNMENT_TO_FINAL
+    expect(() { s1.fontFamily = 'test'; }, throwsA(isNoSuchMethodError));
```
Comment (Hixie) [inline packages/flutter/test/painting/text_style_test.dart]: please put the ignore on the same line as the thing it's ignoring

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 2]
```diff
@@ -120,10 +120,10 @@ Future<Null> benchmarkWidgets(WidgetTesterCallback callback) {
 void expect(dynamic actual, dynamic matcher, {
   String reason,
   bool verbose: false,
-  dynamic formatter
+  @Deprecated("Removed from package:test") dynamic formatter
```
Comment (Hixie) [inline packages/flutter_test/lib/src/widget_tester.dart]: just remove it

File: packages/flutter/test/animation/curves_test.dart [hunk 3]
```diff
@@ -105,35 +105,37 @@ void main() {
   });
 
   test('Invalid transform parameter should assert', () {
-    expect(() => const SawTooth(2).transform(-0.0001), throws);
-    expect(() => const SawTooth(2).transform(1.0001), throws);
+    Matcher isAssertionError = const isInstanceOf<AssertionError>();
```
Comment (Hixie) [inline packages/flutter/test/animation/curves_test.dart]: I would add "throwsAssertionError" and "throwsFlutterError" to flutter_test just to make these more readable

**Comment by Hixie** [comment]
LGTM

**Comment by pq** [comment]
LGTM2!

**Pull request merged**